### ISSUE TITLE
unset last_used_directory when downloads.location.directory is changed

### ISSUE THREAD

**Issue by mehalter** [qutebrowser/qutebrowser#4674]
**Version info (see `:version`)**:

```
qutebrowser v1.6.0
Git commit: 
Backend: QtWebEngine (Chromium 69.0.3497.128)

CPython: 3.7.2
Qt: 5.12.2
PyQt: 5.12.1

sip: 4.19.15
colorama: 0.4.1
pypeg2: 2.15
jinja2: 2.10
pygments: 2.3.1
yaml: 3.13
cssutils: no
attr: 19.1.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.27.2
QtNetwork SSL: OpenSSL 1.1.1b  26 Feb 2019

Style: QFusionStyle
Platform: Linux-5.0.3-arch1-1-ARCH-x86_64-with-arch, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.7/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/micah/.cache/qutebrowser
config: /home/micah/.config/qutebrowser
data: /home/micah/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser
```

**Does the bug happen if you start with `--temp-basedir`?**:

Yes

**Description**

No matter what I do, I cannot get the `downloads.location.directory` setting to be acknowledged. I have tried with or without the download prompt, and my `~/Documents` folder is always used for downloads. I also have `downloads.location.remember` set to `true`.

**How to reproduce**

Set `downloads.location.directory` to a folder, download a file, see where it ends up being downloaded.

**Comment by The-Compiler** [comment]
That's the whole point of `downloads.location.remember` - it'll remember whatever you last selected in the prompt and use that.

**Comment by mehalter** [comment]
Hm, I guess the functionality I am looking for is that if the prompt comes up and I change where I want to download it. That also changes the download location.

Is there any way to have it remember the last location, but I also changing the download location changes the last location?

**Comment by The-Compiler** [comment]
I don't follow. To me, that sounds exactly what `downloads.location.remember` does?

**Comment by mehalter** [comment]
So if I have `downloads.location.remember` set. And say I download something to `~/Documents`, and then I set the `downloads.location.directory` to some other folder, lets say `~/Pictures`. If I try to download something it will still put me at `~/Documents` because I haven't actually downloaded something to `~/Pictures`.

If I don't have `downloads.location.remember` set, changing the `downloads.location.directory` operates correctly. Say I set it to `~/Pictures` and download something the prompt puts me at `~/Pictures`, but if from the prompt I manually move to another folder say `~/Documents`, the next time I download the prompt puts me at `~/Pictures`, not `~/Documents`.

I want the download location to be the previous location that I downloaded, unless I have manually changed the `download.location.directory` since the last time I downloaded.

Does that make sense?
